BranchMonsterFactory/AppDelegate.m

```diff
@@ -8,94 +8,128 @@
+                    [[NSNotificationCenter defaultCenter] postNotificationName:@"pushEditAndViewerViews" object:nil];
+                } else {
+                    self.initialMonster = [self emptyMonster];
+                    [[NSNotificationCenter defaultCenter] postNotificationName:@"pushEditView" object:nil];
+                }
+            } else if(self.foregrounded) {
+                if (receivedMonster) {
+                    self.initialMonster = receivedMonster;
+                    self.foregrounded = NO;
+                    [[NSNotificationCenter defaultCenter] postNotificationName:@"pushEditAndViewerViews" object:nil];
+                }
             }
+            
+        } else {
+            self.initialMonster = [self emptyMonster];
+            [[NSNotificationCenter defaultCenter] postNotificationName:@"pushEditView" object:nil];
         }
-        
-        // launch the next view controller
-        [navController setViewControllers:@[nextVC] animated:YES];
+    
     }];
     
     return YES;
 }
 
+- (BranchUniversalObject *) emptyMonster {
+    
+    
+    BranchUniversalObject*  empty = [[BranchUniversalObject alloc] initWithTitle:@""];
+    [empty setIsMonster];
+    [empty setFaceIndex:0];
+    [empty setBodyIndex:0];
+    [empty setColorIndex:0];
+    [empty setMonsterName:@""];
+     return empty;
+}
+
+
 - (BOOL)application:(UIApplication *)application openURL:(NSURL *)url sourceApplication:(NSString *)sourceApplication annotation:(id)annotation {
-    BOOL wasHandled = [FBAppCall handleOpenURL:url sourceApplication:sourceApplication];
     
     // To receive deep link parameters with the Branch link, you must also call handleDeepLink in the openURL AppDelegate call
-    // This will call the deep link handler block you registered above
-    if (!wasHandled)
-        [[Branch getInstance] handleDeepLink:url];
+    // This will call the deep link handler block registered above
+    [[Branch getInstance] handleDeepLink:url];
+    
+    //other possible code blocks that might want to do something, facebook, etc.
+    //BOOL wasHandled = [FBAppCall handleOpenURL:url sourceApplication:sourceApplication];
+    //if (!wasHandled)...
     
     return YES;
 }
 
-- (void)applicationWillResignActive:(UIApplication *)application {
 
+- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler {
+    BOOL handledByBranch = [[Branch getInstance] continueUserActivity:userActivity];
+    
+    return handledByBranch;
 }
 
-- (void)applicationDidEnterBackground:(UIApplication *)application {
 
+- (void)applicationWillResignActive:(UIApplication *)application {
+}
+
+- (void)applicationDidEnterBackground:(UIApplication *)application {
+    
 }
 
 - (void)applicationWillEnterForeground:(UIApplication *)application {
-
+    NSLog(@"foreground");
+    self.foregrounded = YES;
```

Comment:
not sure what this flag is for, but it worries me that we're using a boolean flag here at all. It seems like this is supposed to prevent the issue I noted above (https://github.com/BranchMetrics/Branch-Example-Deep-Linking-Branchster-iOS/pull/5/files#r47735069), but it unfortunately doesn't